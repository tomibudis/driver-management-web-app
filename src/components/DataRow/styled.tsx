import styled from "styled-components";

export const StyledDataRow = styled.div`
  display: flex;
  flex-direction: column;
  margin: 10px 0px;
  > span:nth-child(1) {
    font-size: 12px;
    opacity: 0.5;
  }

  > span:nth-child(2) {
    font-weight: 500;
  }
`;
