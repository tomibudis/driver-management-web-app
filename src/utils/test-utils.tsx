import React from "react";
import { configureStore, EnhancedStore } from "@reduxjs/toolkit"; // for redux-toolkit
import {
  render as rtlRender,
  RenderOptions,
  RenderResult,
} from "@testing-library/react";
import { ReactElement, ReactNode } from "react";
import { Provider } from "react-redux";
import driversReducer from "../redux/drivers/reducer";

import { RootState } from "../redux/store";
import mockDriversJson from "../views/driver-management/__test__/mock-drivers.json";

type ReduxRenderOptions = {
  preloadedState?: RootState;
  store?: EnhancedStore;
  renderOptions?: Omit<RenderOptions, "wrapper">;
};

function render(
  ui: ReactElement,
  {
    preloadedState = {
      drivers: {
        results: mockDriversJson.results,
        data: mockDriversJson?.results,
      },
    },
    store = configureStore({
      reducer: { drivers: driversReducer },
      preloadedState,
    }),
    ...renderOptions
  }: ReduxRenderOptions = {}
): RenderResult {
  function Wrapper({ children }: { children?: ReactNode }): ReactElement {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";
export { render };
