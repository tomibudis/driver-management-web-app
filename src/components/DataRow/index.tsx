import React from "react";

import { StyledDataRow } from "./styled";

interface DataRowProps {
  label: string;
  value: string;
  isDate?: boolean;
}
const DataRow: React.FC<DataRowProps> = ({ value, label }) => {
  return (
    <StyledDataRow>
      <span>{label}</span>
      <span>{value}</span>
    </StyledDataRow>
  );
};

export default DataRow;
