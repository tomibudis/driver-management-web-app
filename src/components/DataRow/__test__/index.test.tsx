import React from "react";
import { render, screen } from "~utils/test-utils";

import DataRow from "../index";

describe("<DataRow />", () => {
  it("render data row component", () => {
    render(<DataRow label="test label" value="example value" />);

    const children = screen.getByText("example value");
    expect(children).toBeInTheDocument();
  });
  it("render label correctly", () => {
    render(<DataRow label="test label" value="example value" />);
    const labelCard = screen.getByText("test label");
    expect(labelCard).toBeInTheDocument();
  });
  it("matched with a snapshot", () => {
    const { container } = render(
      <DataRow label="test label" value="example value" />
    );
    expect(container).toMatchSnapshot();
  });
});
