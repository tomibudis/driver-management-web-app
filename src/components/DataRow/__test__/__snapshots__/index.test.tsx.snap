// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataRow /> matched with a snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ eYrFET"
  >
    <span>
      test label
    </span>
    <span>
      example value
    </span>
  </div>
</div>
`;
